src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -496,13 +496,24 @@ ParsedQuery Visitor::visit(Parser::ModifyContext* ctx) {
       return true;
     }
   };
+  auto isVisibleIfVariableGraph =
+      [this](const SparqlTripleSimpleWithGraph::Graph& graph) {
+        if (std::holds_alternative<Variable>(graph)) {
+          return std::ranges::find(parsedQuery_.getVisibleVariables(),
+                                   std::get<Variable>(graph)) !=
+                 parsedQuery_.getVisibleVariables().end();
```

Comment:
1. This common code can be factored out into a separate lambda
2. It can maybe use `ad_utility::contains` to reduce the verbosity with the explicit comparison to `end()`.